lib/graphql/dataloader/source.rb

```diff
@@ -76,17 +90,24 @@ def pending?
       # @api private
       # @return [void]
       def run_pending_keys
+        if @fetching_keys.any?
+          @pending_keys -= @fetching_keys
+        end
         return if @pending_keys.empty?
         fetch_keys = @pending_keys.uniq
+        @fetching_keys.concat(fetch_keys)
         @pending_keys = []
         results = fetch(fetch_keys)
         fetch_keys.each_with_index do |key, idx|
           @results[key] = results[idx]
         end
+        nil
       rescue StandardError => error
         fetch_keys.each { |key| @results[key] = error }
       ensure
-        nil
+        if fetch_keys
```

Comment:
It seems `fetch_keys` is always an instance of Array because it's the result of  `@pending_keys.uniq` and never falsy

Reply:
Ah, yes. You're right. Never mind, sorry.